fe/src/main/java/org/apache/doris/load/loadv2/SparkLoadJob.java

```diff
@@ -0,0 +1,249 @@
+    private Set<Long> finishedReplicas = Sets.newHashSet();
+    private Set<Long> quorumTablets = Sets.newHashSet();
+    private Set<Long> fullTablets = Sets.newHashSet();
+
+    // only for log replay
+    public SparkLoadJob() {
+        super();
+        jobType = EtlJobType.SPARK;
+    }
+
+    public SparkLoadJob(long dbId, String label, ResourceDesc resourceDesc, OriginStatement originStmt)
+            throws MetaNotFoundException {
+        super(dbId, label, originStmt);
+        this.resourceDesc = resourceDesc;
+        timeoutSecond = Config.spark_load_default_timeout_second;
+        jobType = EtlJobType.SPARK;
+    }
+
+    public String getHiveTableName() {
+        return hiveTableName;
+    }
+
+    @Override
+    protected void setJobProperties(Map<String, String> properties) throws DdlException {
+        super.setJobProperties(properties);
+
+        // set spark resource and broker desc
+        setResourceInfo();
+
+        // global dict
+        if (properties != null) {
+            if (properties.containsKey(BITMAP_DATA_PROPERTY)) {
+                hiveTableName = properties.get(BITMAP_DATA_PROPERTY);
+            }
+        }
+    }
+
+    /**
+     * merge system conf with load stmt
+     * @throws DdlException
+     */
+    private void setResourceInfo() throws DdlException {
+        // spark resource
+        String resourceName = resourceDesc.getName();
+        Resource oriResource = Catalog.getCurrentCatalog().getResourceMgr().getResource(resourceName);
+        if (oriResource == null) {
+            throw new DdlException("Resource does not exist. name: " + resourceName);
+        }
+        Preconditions.checkState(oriResource instanceof SparkResource);
+        sparkResource = ((SparkResource) oriResource).getCopiedResource();
+        sparkResource.update(resourceDesc);
+
+        // broker desc
+        Map<String, String> brokerProperties = sparkResource.getBrokerPropertiesWithoutPrefix();
+        brokerDesc = new BrokerDesc(sparkResource.getBroker(), brokerProperties);
+    }
+
+    /**
+     * load job already cancelled or finished, clear job below:
+     * 1. kill etl job and delete etl files
+     * 2. clear push tasks and infos that not persist
+     */
+    private void clearJob() {
+        Preconditions.checkState(state == JobState.FINISHED || state == JobState.CANCELLED);
+
+        LOG.debug("kill etl job and delete etl files. id: {}, state: {}", id, state);
+        // TODO(wyb): spark-load
+        //SparkEtlJobHandler handler = new SparkEtlJobHandler();
+        if (state == JobState.CANCELLED) {
+            if ((!Strings.isNullOrEmpty(appId) && sparkResource.isYarnMaster()) || sparkAppHandle != null) {
+                try {
+                    // TODO(wyb): spark-load
+                    //handler.killEtlJob(sparkAppHandle, appId, id, sparkResource);
+                } catch (Exception e) {
+                    LOG.warn("kill etl job failed. id: {}, state: {}", id, state, e);
```

Comment:
Save the error msg somewhere for user to get?

Reply:
I think it’s not necessary, because clear job is just trying to kill etl job as much as possible.